administrator/components/com_media/Model/ApiModel.php

```diff
@@ -129,6 +145,11 @@ public function getFiles($adapter, $path = '/', $options = array())
 		// Add adapter prefix to all the files to be returned
 		foreach ($files as $file)
 		{
+			if ($file->type == 'file' && !$this->isMediaFile($file->path))
+			{
+				continue;
```

Comment:
This will not remove the files from json response!
 
We should unset them or not get them from filesystem (see comment above).